src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,70 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(movieSession);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException(String.format(
+                    "Can't save movie session: %s", movieSession), e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movieSession;
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return Optional.ofNullable(session.get(MovieSession.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException(String.format(
+                    "Can't find movie session with id: %d", id), e);
+        }
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDateTime fromTime,
+                                                    LocalDateTime toTime) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> findAvailableSessionsQuery = session.createQuery(
+                    "from MovieSession ms "
+                            + "left join fetch ms.movie m "
+                            + "where m.id= :id "
+                            + "and ms.showTime "
+                            + "between :fromTime and :toTime", MovieSession.class);
+            findAvailableSessionsQuery.setParameter("id", movieId);
+            findAvailableSessionsQuery.setParameter("fromTime", fromTime);
+            findAvailableSessionsQuery.setParameter("toTime", toTime);
+            return findAvailableSessionsQuery.getResultList();
```

Comment:
The method `findAvailableSessions` should not fetch all `MovieSession` and then filter them in the service layer. Instead, it should directly extract the correct sessions for the given day from the database. The query should filter by date, not by the `fromTime` and `toTime` `LocalDateTime` objects, to ensure that only sessions for the specific day are returned.